docs/unit-testing-for-monitoring-metrics.md

```diff
@@ -10,42 +10,56 @@ The ARO monitor component (the part of the aro binary you activate when you exec
 
 There are two ways to set to acchieve this: 
-- Run the Geneva MDM container locally (won't work on macOS, see Remote Container section below)
+- Run the Geneva MDM container locally
 - Spawn a VM, start the Geneva container there and connect/tunnel to it.
 
-### Local Container Setup
+and two protocols to chose from:
+- Unix Domain Sockets, which is the way production is currently (April 2022) run
+- or UDP, which is much easier to use and is the way it will be used on kubernetes clusters in the future
+
+## Local Container Setup
 
 Before you start, make sure :
 - to run `source ./env`
 - you ran `SECRET_SA_ACCOUNT_NAME=rharosecrets make secrets` before 
 - know which "account" and "namespace" value you want to use on Geneva INT for your metric data and
   update your env to set the 
   - CLUSTER_MDM_ACCOUNT
-  - CLUSTER_MDM_NAMESPACE
-   
+  - CLUSTER_MDM_NAMESPACE variables before you start the monitor. 
+
+ The container needs to be provided with the Geneva key and certificate. For the INT instance that is the rp-metrics-int.pem you find in the secrets folder after running the `make secrets` command above.  
+
+### The Unix Domain Socket way
+
+An example docker command to start the container locally using UNIX domain sockets is here (you may need to adapt some parameters):
+[Example](../hack/local-monitor-testing/samples/dockerStartCommand-UDS.sh). As mentioned this is the way it's run in production. The script will configure the mdm container to connect to Geneva INT.  The sample script tries to copy the certificate file to /etc/mdm.pem to mimic production. You can change that by changing the corresponding volume definition. In that case you'll also need to set the `ARP_STATDS_SOCKET` environment variable to `"unix:/<path to where the socket file should be created>"` to point the monitor (or RP for that matter) to the same place/file.
+
+When you start the montitor locally in local dev mode, by default he monitor looks for a Unix Socket file called `mdm_statsd.socket` in the current directory from where the cmd/aro command is executed, typically either the base folder of your repo or `./cmd/aro` is started through VSCode (for example)
+
+> :warning For macOS users: Neither Docker For Desktop nor podman machine currently support Unix Domain Sockets, so this will not work for you. Refer to the Remote Container Setup instead or use the UDP way.
+
+### The UDP way
+
 
-  variables before you start the monitor. 
+An example docker command to start the container locally using UDP is here (you may need to adapt some parameters):
+[Example](../hack/local-monitor-testing/samples/dockerStartCommand-UDP.sh). This will start an mdm container on `0.0.0.0:8125`. 
+Please note that you need to set the `ARP_STATDS_SOCKET` environment variable at monitor start up time so that the monitor finds that socket.
```

Comment:
I don't see references to `ARP_STATDS_SOCKET` in the codebase. What should this be set to for the "UDP way".

Reply:
That's a typo combined with a failed find/replace. The name is now MDM_STATSD_SOCKET.  Commit is on it's way